packages/compass-e2e-tests/helpers/db-logger.js

```diff
@@ -0,0 +1,272 @@
+// @ts-check
+const Mocha = require('mocha');
+
+const debug = require('debug')('compass-e2e-tests:db-logger');
+
+const DB_NAME = 'compass_e2e';
+const COLLECTION_NAME = 'results';
+
+const {
+  EVENT_HOOK_BEGIN,
+  EVENT_HOOK_END,
+  EVENT_TEST_BEGIN,
+  EVENT_TEST_FAIL,
+  EVENT_TEST_PASS
+} = Mocha.Runner.constants;
+
+// env vars to store with the metrics. Be careful not to include secrets.
+const ENV_VARS = [
+  'CI',
+
+  // evergreen
+  'EVERGREEN_AUTHOR',
+  'EVERGREEN_BRANCH_NAME',
+  'EVERGREEN_BUILD_ID',
+  'EVERGREEN_BUILD_VARIANT',
+  'EVERGREEN_EXECUTION',
+  'EVERGREEN_IS_PATCH',
+  'EVERGREEN_PROJECT',
+  'EVERGREEN_REVISION',
+  'EVERGREEN_TASK_ID',
+  'EVERGREEN_TASK_NAME',
+  'EVERGREEN_TASK_URL',
+  'EVERGREEN_VERSION_ID',
+  'EVERGREEN_WORKDIR',
+
+  // github
+  'GITHUB_WORKFLOW',
+  'GITHUB_RUN_ID',
+  'GITHUB_RUN_NUMBER',
+  'GITHUB_JOB',
+  'GITHUB_ACTION',
+  'GITHUB_ACTION_PATH',
+  'GITHUB_ACTIONS',
+  'GITHUB_ACTOR',
+  'GITHUB_REPOSITORY',
+  'GITHUB_EVENT_NAME',
+  'GITHUB_EVENT_PATH',
+  'GITHUB_WORKSPACE',
+  'GITHUB_SHA',
+  'GITHUB_REF',
+  'GITHUB_HEAD_REF',
+  'GITHUB_BASE_REF',
+  'GITHUB_SERVER_URL',
+  'GITHUB_API_URL',
+  'GITHUB_GRAPHQL_URL',
+  'RUNNER_NAME',
+  'RUNNER_OS',
+  'RUNNER_TEMP',
+  'RUNNER_TOOL_CACHE',
+];
+
+function joinPath(parts) {
+  // Turn an array of test/hook path components into a string we can use as an
+  // identifier for the test, hook or suite
+
+  return parts.join(' / ');
+}
+
+function githubWorkflowRunUrl() {
+  const serverURL = process.env.GITHUB_SERVER_URL;
+  const repository = process.env.GITHUB_REPOSITORY;
+  const runID = process.env.GITHUB_RUN_ID;
+
+  return `${serverURL}/${repository}/actions/runs/${runID}`;
+}
+
+class DBLogger {
+  constructor(client, runner) {
+    this.client = client;
+    const db = this.client.db(DB_NAME);
+    this.collection = db.collection(COLLECTION_NAME);
+
+    this.context = {};
+
+    // copy known env vars as-is if they are set
+    this.context.env = {};
+    for (const name of ENV_VARS) {
+      const value = process.env[name];
+      if (value) {
+        this.context.env[name] = value;
+      }
+    }
+
+    // infer some common variables
+
+    this.context.ci = process.env.EVERGREEN ? 'evergreen'
+      : process.env.GITHUB_ACTIONS ? 'github-actions'
+      : 'unknown';
+
+    this.context.platform = process.platform;
+
+    this.context.author = process.env.EVERGREEN_AUTHOR || process.env.GITHUB_ACTOR || 'unknown';
```

Comment:
Something here I'm unsure about - should I leave these out, set them to undefined or null? Part of me wants to differentiate between values we didn't know at the time of inserting the data vs fields we only started to store later but I might be overthinking this :)

What's best practise - should a field in a collection always have the same type or is it OK for it to be a string sometimes and, say, null others?